cmd/syncthing/connections.go

```diff
@@ -29,7 +39,44 @@ type connectionSvc struct {
 	myID   protocol.DeviceID
 	model  *model.Model
 	tlsCfg *tls.Config
-	conns  chan *tls.Conn
+	conns  chan connection
+
+	lastRelayCheck map[protocol.DeviceID]time.Time
+
+	mut      sync.RWMutex
+	connType map[protocol.DeviceID]connectionType
+}
+
+type connection struct {
+	conn     *tls.Conn
+	connType connectionType
+}
+
+const (
+	connectionTypeBasicAccept connectionType = 1 << iota
```

Comment:
bit shiftery implies something could be `connectionTypeBasicDial | connectionTypeRelayAccept` or something which does not seem to be the case?